components/PlantForm.js

```diff
@@ -2,38 +2,40 @@ import styled from "styled-components";
 import { uid } from "uid";
 import { useRouter } from "next/navigation";
 
-export default function PlantForm({ onAddPlants }) {
+export default function PlantForm({ onSubmit, $isEdit, editPlant }) {
   const router = useRouter();
   function handleSubmit(event) {
     event.preventDefault();
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
     const randomId = uid();
     const newPlant = {
-      id: `${randomId}`,
+      id: $isEdit ? editPlant?.id : `${randomId}`,
```

Comment:
```suggestion
      id: $isEdit ? editPlant.id : `${randomId}`,
```

If you're already guarding with the `isEdit`, the optional chaining is redundant and might actually hide bugs.